src/components/Header/HeaderNav/HeaderNav.js

```diff
@@ -1,31 +1,54 @@
 import React from 'react';
 
-const HeaderNav = (props) => {
-  const navItems = props.nav;
-  return (
-    <nav>
-      <ul>
-        {navItems.map((item, index) => (
-          <li key={index}>
-            <a className={item.isSelected ? 'active' : ''} href={item.url}>
-              {item.title}
-            </a>
-            {item.subNav ? (
-              <ul>
-                {item.subNav.map((itemSubNav, index) => (
-                  <li key={index}>
-                    <a className={itemSubNav.isSelected ? 'active' : ''} href={itemSubNav.url}>
-                      {itemSubNav.title}
-                    </a>
-                  </li>
-                ))}
-              </ul>
-            ) : null}
-          </li>
-        ))}
-      </ul>
-    </nav>
-  );
-};
+class HeaderNav extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.hoverMenu = this.hoverMenu.bind(this);
+  }
+
+  hoverMenu(hasSubNav) {
+    if (Array.isArray(hasSubNav) && hasSubNav.length) {
```

Comment:
If I understand it fine, here you are overloading the meaning of _hasSubNav_ with toggle status, what do you think about simply pass a boolean parameter?

Also, maybe we could break the components in a different way to avoid to verify _hasSubNav_ here.